tests/test_core_to_contrib.py

```diff
@@ -16,787 +16,819 @@
+        ),
+        (
+            "airflow.gcp.sensors.cloud_storage_transfer_service."
+            "GCPTransferServiceWaitForJobStatusSensor",
+            "airflow.contrib.sensors.gcp_transfer_sensor."
+            "GCPTransferServiceWaitForJobStatusSensor",
+        ),
+        (
+            "airflow.gcp.sensors.pubsub.PubSubPullSensor",
+            "airflow.contrib.sensors.pubsub_sensor.PubSubPullSensor",
+        ),
+        (
+            "airflow.gcp.sensors.bigquery.BigQueryTableSensor",
+            "airflow.contrib.sensors.bigquery_sensor.BigQueryTableSensor",
+        ),
+        (
+            "airflow.gcp.sensors.gcs.GoogleCloudStorageObjectSensor",
+            "airflow.contrib.sensors.gcs_sensor.GoogleCloudStorageObjectSensor",
+        ),
+        (
+            "airflow.gcp.sensors.gcs.GoogleCloudStorageObjectUpdatedSensor",
+            "airflow.contrib.sensors.gcs_sensor.GoogleCloudStorageObjectUpdatedSensor",
+        ),
+        (
+            "airflow.gcp.sensors.gcs.GoogleCloudStoragePrefixSensor",
+            "airflow.contrib.sensors.gcs_sensor.GoogleCloudStoragePrefixSensor",
+        ),
+        (
+            "airflow.gcp.sensors.gcs.GoogleCloudStorageUploadSessionCompleteSensor",
+            "airflow.contrib.sensors.gcs_sensor.GoogleCloudStorageUploadSessionCompleteSensor",
+        ),
+    ]
+
     def assert_is_subclass(self, clazz, other):
-        self.assertTrue(issubclass(clazz, other))
+        self.assertTrue(issubclass(clazz, other), "{} is not subclass of {}".format(clazz, other), )
 
-    def assert_proper_import(self, new_resource: str, old_resource: str):
-        old_path, _, old_class_name = old_resource.rpartition(".")
-        new_path, _, new_class_name = new_resource.rpartition(".")
-        with self.assertWarns(DeprecationWarning):
+    def assert_proper_import(self, old_resource):
+        deprecation_warning_msg = \
+            "This module is deprecated."
+        old_path, _, _ = old_resource.rpartition(".")
+        with self.assertWarnsRegex(DeprecationWarning, deprecation_warning_msg) as warning_msg:
             # Reload to see deprecation warning each time
-            old_module = importlib.reload(importlib.import_module(old_path))
-        new_module = importlib.import_module(new_path)
+            importlib.reload(importlib.import_module(old_path))
+        self.assertTrue(deprecation_warning_msg, warning_msg)
+
+    def get_class_from_path(self, path_to_class):
+        path, _, class_name = path_to_class.rpartition(".")
+        module = importlib.import_module(path)
+        class_ = getattr(module, class_name)
+        return class_
+
+    @parameterized.expand(MODULE)
+    def test_is_class_deprecated(self, new_module, old_module):
+        old_module_class = self.get_class_from_path(old_module)
+        with self.assertWarns(DeprecationWarning):
+            with mock.patch("{}.__init__".format(new_module)) as init_mock:
+                old_module_class()
```

Comment:
We want to assert that the deprecation message includes information about new path to class.

Reply:
Done